src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Dependency.cs

```diff
@@ -196,13 +197,13 @@ protected set
         public bool Implicit { get; private set; }
         public bool Visible { get; }
 
-        // Not readonly because SetProperties needs to mutate the clone.
-        private DependencyIconSet _iconSet;
 
-        public ImageMoniker Icon => _iconSet.Icon;
-        public ImageMoniker ExpandedIcon => _iconSet.ExpandedIcon;
-        public ImageMoniker UnresolvedIcon => _iconSet.UnresolvedIcon;
-        public ImageMoniker UnresolvedExpandedIcon => _iconSet.UnresolvedExpandedIcon;
+        public ImageMoniker Icon => IconSet.Icon;
+        public ImageMoniker ExpandedIcon => IconSet.ExpandedIcon;
+        public ImageMoniker UnresolvedIcon => IconSet.UnresolvedIcon;
+        public ImageMoniker UnresolvedExpandedIcon => IconSet.UnresolvedExpandedIcon;
```

Comment:
Are these needed now when we added IconSet as public property?

Reply:
`Dependency` implements `IDependencyModel` which is a public interface, so unfortunately yes.